src/openms/source/ANALYSIS/TARGETED/MetaboTargetedAssay.cpp

```diff
@@ -0,0 +1,498 @@
+      {
+        for (auto spec_it = transition_spectrum.begin(); spec_it != transition_spectrum.end(); ++spec_it)
+        {
+          if (transition_spectrum.getPrecursors()[0].getMZ() == spec_it->getMZ())
+          {
+            transition_spectrum.erase(spec_it);
+            break;
+          }
+        }
+      }
+
+      // find max and min intensity peak
+      max_int = max_element(transition_spectrum.begin(), transition_spectrum.end(), intensityLess_)->getIntensity();
+      min_int = min_element(transition_spectrum.begin(), transition_spectrum.end(), intensityLess_)->getIntensity();
+
+      // no peaks or all peaks have same intensity (single peak / noise)
+      if (min_int >= max_int)
+      {
+        continue;
+      }
+
+      vector <TargetedExperimentHelper::RetentionTime> v_cmp_rt;
+      TargetedExperimentHelper::RetentionTime cmp_rt;
+      cmp_rt.setRT(feature_rt);
+      v_cmp_rt.push_back(cmp_rt);
+      cmp.rts = v_cmp_rt;
+
+      description = ListUtils::concatenate(v_description, ",");
+      cmp.id = String(transition_group_counter) + "_" + description + "_" + file_counter;
+      cmp.setMetaValue("CompoundName", description);
+      cmp.smiles_string = "NA";
+
+      sumformula = ListUtils::concatenate(v_sumformula, ",");
+      cmp.molecular_formula = sumformula;
+
+      adduct = ListUtils::concatenate(v_adduct, ",");
+      cmp.setMetaValue("Adducts", adduct);
+
+      // threshold should be at x % of the maximum intensity
+      // hard minimal threshold of min_int * 1.1
+      float threshold_transition = max_int * (transition_threshold / 100);
+      float threshold_noise = min_int * 1.1;
+
+      int transition_counter = 0;
+      // here ms2 spectra information is used
+      for (auto spec_it = transition_spectrum.begin(); spec_it != transition_spectrum.end(); ++spec_it)
+      {
+        ReactionMonitoringTransition rmt;
+        rmt.clearMetaInfo();
+
+        float current_int = spec_it->getIntensity();
+        double current_mz = spec_it->getMZ();
+
+        // write row for each transition
+        // current int has to be higher than transition thresold and should not be smaller than threshold noise
+        if (current_int > threshold_transition && current_int > threshold_noise)
+        {
+          float rel_int = current_int / max_int;
+
+          rmt.setPrecursorMZ(highest_precursor_mz);
+          rmt.setProductMZ(current_mz);
+          rmt.setLibraryIntensity(rel_int);
+
+          description = ListUtils::concatenate(v_description, ",");
+          rmt.setCompoundRef (String(transition_group_counter) + "_" + description + "_" + file_counter);
+          rmt.setNativeID (String(transition_group_counter)+ "_" + String(transition_counter)+ "_" + description + "_" + file_counter);
+
+          v_rmt.push_back(rmt);
+          transition_counter += 1;
+        }
+      }
+      transition_group_counter += 1;
+      MetaboTargetedAssay mta;
+      mta.precursor_mz = highest_precursor_mz;
```

Comment:
it looks like this line and line 294 just store the same information

Reply:
Yes, the precursor_mz is stored in the ReactionMonitoringTransition as well. 
It is not needed later on - will remove it. 

As far as I see it I could potentially remove precursor_mz, precursor_rt, precursor_charge.
The precursor_intensity, compound_name and the compound_adduct are later on used for sorting and unique filtering. 

What do you think?

Reply:
Yes, I suggest to remove duplicate entries since its confusing that its in two places -- and think of   potentially even move some info into Compound itself if you need it often